File: 1018.py
def func(x, y):
    n = 0
    for i in range(y, y+8):
        for j in range(x, x+8):
            if (i+j)%2==0:
                if s[i][j] == 'W': n += 1
            elif (i+j)%2==1:
                if s[i][j] == 'B': n += 1
    return min(n, (64 - n))

n, m = map(int, input().split())
a = [0 for i in range(60)]
s = [[0 for i in range(60)] for i in range(60)]
for i in range(n):
    a = input()
    s[i] = list(a) + [0] * 60
k = 64
for i in range(m-7):
    for j in range(n-7):
        if k > func(i, j):
            k = func(i, j)
print(k)

File: 1018.c
#include <stdio.h>

int main() {
    int n, m;
    scanf("%d %d", &n, &m);
    int min = 9999, x = 0, y = 0;
    char a[100][100] = {{0, }, };
    for (int i = 0; i < n; i++) {
        scanf("%s", a[i]);
    }
    
    for (int i = 0; i <= n - 8; i++) {
        for (int j = 0; j <= m - 8; j++) {
            x = 0;
            for (int k = 0; k < 8; k++) {
                for (int l = 0; l < 8; l++) {
                    if ((i + j + k + l) % 2 == 0 && a[i + k][j + l] == 'B')
                        x++;
                    else if ((i + j + k + l) % 2 == 1 && a[i + k][j + l] == 'W')
                        x++;
                }
            }
            if (x > 32)
                x = 64 - x;
            if (min > x)
                min = x;
        }
    }
    
    printf("%d", min);
}
